bgpd/bgp_nb_config.c

```diff
@@ -332,33 +669,62 @@ int routing_control_plane_protocols_control_plane_protocol_bgp_global_route_refl
 int routing_control_plane_protocols_control_plane_protocol_bgp_global_route_reflector_allow_outbound_policy_modify(
 	struct nb_cb_modify_args *args)
 {
-	switch (args->event) {
-	case NB_EV_VALIDATE:
-	case NB_EV_PREPARE:
-	case NB_EV_ABORT:
-	case NB_EV_APPLY:
-		/* TODO: implement me. */
-		break;
-	}
+	if (args->event != NB_EV_APPLY)
+		return NB_OK;
+
+	struct bgp *bgp;
+	int ret;
+
+	bgp = nb_running_get_entry(args->dnode, NULL, true);
+
+	if (yang_dnode_get_bool(args->dnode, NULL))
+		SET_FLAG(bgp->flags, BGP_FLAG_RR_ALLOW_OUTBOUND_POLICY);
+	else
+		UNSET_FLAG(bgp->flags, BGP_FLAG_RR_ALLOW_OUTBOUND_POLICY);
+
+	update_group_announce_rrclients(bgp);
+
+	ret = bgp_clear_star_soft_out(bgp->name, args->errmsg,
+				      args->errmsg_len);
```

Comment:
update_group_announce_rrclients and bgp_clear_star_soft_out should be only called if the flag has toggeled.

Reply:
The northbound callback is only called when the flag is flipped or configuration has changed.